source/controller/party.js

```diff
@@ -19,29 +19,129 @@ class Party {
     return res.status(200).json(response);
   }
 
-    static createParty(req, res) {
-      const {
-        name, logoUrl, hqAddress
-      } = req.body;
-
-      let id = data[0].length + 100;
-      let newParty = {
-        id,
-        name,
-        hqAddress,
-        logoUrl,
+  static createParty(req, res) {
+    const {
+      name, logoUrl, hqAddress
+    } = req.body;
+
+    let id = data[0].length + 100;
+    let newParty = {
+      id,
+      name,
+      hqAddress,
+      logoUrl,
+    }
+    data[0].push(newParty);
+    let response = {
+      status : 201,
+      data : [ {
+      id,
+      name,
+      } ]
+    };
+
+    return res.status(201).json(response);
+  }
+
+  static async getOneParty(req, res) {
+    let { id } = req.params;
+    id = parseInt(id);
+    if (isNaN(id)) {
+      return res.status(400).json({
+        status: 400,
+        error: 'Invalid party Id'
+      });
+    }
+    let parties = data[0];
+    let foundParty = false;
+    await parties.forEach((party) => {
+      if(party.id === id){
+        foundParty = true;
+        let oneParty = {
+          id: party.id,
+          name: party.name,
+          logoUrl: party.logoUrl,
+        }
+        let response = {
+          status: 200 ,
+          data: [oneParty],
+        }
+        return res.status(200).json(response);
       }
-      data[0].push(newParty);
-      let response = {
-        status : 201,
-        data : [ {
-        id,
-        name,
-        } ]
-      };
-
-      return res.status(201).json(response);
+    });
+    if(!foundParty){
+      return res.status(404).json({
+        status: 404,
+        error: 'That party could not be found'
+      });
+    } 
+  }
+
+  static async editParty(req, res) {
+    const { id } = req.params;
+    const {
+      name,
+    } = req.body;
+    
+    let parties = data[0];
+    let foundParty = false;
```

Comment:
I think this variable isn't useful. If party.id === id at some point, you will return the response right? So if the function ever gets to line 103, you can go ahead to return res.status(XX).json.

What do you think?